### ISSUE TITLE
Typography for simple paragraphs

### ISSUE THREAD

**Issue by stevesims** [govuk-react/govuk-react#503]
govuk-frontend has several different types of paragraph styles as can be seen here:
https://design-system.service.gov.uk/styles/typography/#paragraphs

Right now we support a (simple) LeadParagraph component which is styled to match govuk-frontend's `govuk-body-l`, and then have a separate Paragraph component that supports Markdown markup for content.

We don't have support for simple paragraphs with different base font sizes, as per govuk-frontend's `govuk-body-m` and `govuk-body-s` styles.  Alternatively the Markdown-supporting paragraph also does not support specifying different base font sizes.

**Comment by stevesims** [comment]
Adding a `SimpleParagraph` component supporting a font size indicator would be sensible.

The current `LeadParagraph` would become a simple wrapper for `SimpleParagraph`, setting appropriate props.

The existing `Paragraph` component can keep support for markdown - but it should also add support for setting different base font sizes, consistent with `SimpleParagraph`.

**Comment by stevesims** [comment]
It may instead be better to remove markdown support from paragraph.  Markdown support can be moved to a separate package.

`Paragraph` would then become a simpler component that supports the same font size options as the govuk-frontend paragraph.

`LeadParagraph` would be deprecated

**Comment by penx** [comment]
agreed Paragraph component should match govuk-frontend and current component moved to MarkDownParagraph or similar

**Comment by penx** [comment]
@stevesims I would prefer that markdown support is out of scope for govuk-react for now, given the edge cases it creates and maintenance overhead.

What do you think?

**Comment by penx** [comment]
Also see #223

**Comment by domdomegg** [comment]
This would help me in my applications where I don't want whatever is being shown on screen to be interpreted as markdown, particularly if it's user generated content (to avoid users putting in misleading page elements like links) or if I want to add links that should open in a new tab (can workaround with a custom linkRenderer, but fiddly particularly if want some links that do and some that don't open in a new tab in the same paragraph). It might also improve page performance, particularly where I know it's just going to be static text with no features anyways.

Potentially adding MarkdownParagraph as an alias for Paragraph, and encouraging users to prefer using MarkdownParagraph would help make it more explicit in the code as to what it's doing and allow us to eventually deprecate that name it and reuse the same to align with GDS.
